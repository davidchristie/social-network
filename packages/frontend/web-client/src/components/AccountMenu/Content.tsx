import { Avatar, Menu } from "design-system";
import React from "react";
import ButtonLink from "../ButtonLink";
import LogoutButton from "../LogoutButton";

interface Account {
  profile: {
    avatar: {
      url: string;
    } | null;
    id: string;
  };
}

export interface Props {
  account: Account;
}

interface State {
  anchorElement: HTMLElement | null;
}

export default class Content extends React.Component<Props, State> {
  public state = {
    anchorElement: null,
  };

  public render () {
    const { account } = this.props;
    return (
      <div>
        {this.renderAvatar(account)}
        {this.renderMenu(account)}
      </div>
    );
  }

  private renderAccountButton = () => {
    return (
      <ButtonLink
        onClick={this.handleClose}
        to="/account"
      >
        Account
      </ButtonLink>
    );
  }

  private renderProfileButton = (account: Account) => {
    return (
      <ButtonLink
        onClick={this.handleClose}
        to={`profile/${account.profile.id}`}
      >
        Profile
      </ButtonLink>
    );
  }

  private handleClick = event => {
    console.log(event.currentTarget);
    this.setState({
      anchorElement: event.currentTarget,
    });
  }

  private handleClose = () => {
    this.setState({
      anchorElement: null,
    });
  }

  private renderAvatar = (account: Account) => {
    const { profile: { avatar } } = account;
    const image = avatar ? avatar.url : undefined;
    return (
      <Avatar
        image={image}
        onClick={this.handleClick}
        size="small"
      />
    );
  }

  private renderMenu = (account: Account) => {
    return (
      <Menu
        anchorElement={this.state.anchorElement}
        onClose={this.handleClose}
        open={Boolean(this.state.anchorElement)}
      >
        {this.renderProfileButton(account)}
        {this.renderAccountButton()}
        <hr />
        <LogoutButton />
      </Menu>
    );
  }
}
