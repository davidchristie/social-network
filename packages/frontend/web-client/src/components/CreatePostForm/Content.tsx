import { Button, Section, TextArea } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import { CreatePostData, CreatePostVariables } from "../../mutations/CreatePost";

export interface Props {
  createPost: MutationFn<CreatePostData, CreatePostVariables>;
}

export interface State {
  text: string;
}

export default class Content extends React.Component<Props, State> {
  public state: State = {
    text: "",
  };

  public render () {
    return (
      <Section>
        <form onSubmit={this.handleFormSubmitted}>
          <div>
            <TextArea
              name="text"
              onChange={this.handleTextChanged}
              required={true}
              value={this.state.text}
            />
          </div>
          <Button type="submit">Post</Button>
        </form>
      </Section>
    );
  }

  private handleFormSubmitted = async (event: React.FormEvent) => {
    event.preventDefault();
    await this.props.createPost({
      variables: {
        text: this.state.text,
      },
    });
    this.setState({
      text: "",
    });
  }

  private handleTextChanged = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
    this.setState({
      text: event.target.value,
    });
  }
}
