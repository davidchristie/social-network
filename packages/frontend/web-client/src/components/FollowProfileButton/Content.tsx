import { Button } from "design-system";
import React from "react";

export interface Props {
  account: {
    profile: {
      following: Array<{
        id: string;
      }>;
    };
  };
  profileId: string;
}

const Content: React.StatelessComponent<Props> = ({ account, profileId }) => {
  const isFollowing = account.profile.following.some(followed => {
    return followed.id === profileId;
  });
  return isFollowing
    ? <Button>Unfollow</Button>
    : <Button>Follow</Button>;
};

export default Content;
