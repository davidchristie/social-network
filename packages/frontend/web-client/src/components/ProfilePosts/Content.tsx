import React from "react";
import { PostSummaryData } from "../../fragments/PostSummary";
import PostSummary from "../PostSummary";

interface Props {
  profile: {
    posts: PostSummaryData[]
  };
}

const Content: React.StatelessComponent<Props> = ({ profile }) => {
  return (
    <div className="ProfilePosts">
      {
        profile.posts
          .map(post => <PostSummary key={post.id} post={post} />)
          .reverse()
      }
    </div>
  );
};

export default Content;
