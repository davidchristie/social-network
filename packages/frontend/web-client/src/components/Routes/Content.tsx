import React from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import Page from "../Page";
import {
  defaultAuthenticatedPath,
  defaultUnauthenticatedPath,
  routes
} from "./constants";

export interface Props {
  authenticated: boolean;
}

const Content: React.StatelessComponent<Props> = ({ authenticated }) => {
  const defaultPath = authenticated
    ? defaultAuthenticatedPath
    : defaultUnauthenticatedPath;
  const filteredRoutes = routes
    .filter(route => {
      return authenticated
        ? !route.onlyUnauthenticated
        : !route.onlyAuthenticated;
    });
  return (
    <Switch>
      {filteredRoutes.map(route => {
        return (
          <Route
            exact={true}
            key={route.path}
            path={route.path}
            render={() => <Page load={route.load} />}
          />
        );
      })}
      <Route
        path="*"
        render={() => <Redirect to={defaultPath} />}
      />
    </Switch>
  );
};

export default Content;
