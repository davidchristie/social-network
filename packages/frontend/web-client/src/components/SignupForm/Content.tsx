import { Alert, Button, Input, Section } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import { Link } from "react-router-dom";
import { SignupData, SignupVariables } from "../../mutations/Signup";

export interface Props {
  signup: MutationFn<SignupData, SignupVariables>;
}

export interface State {
  email: string;
  errorMessage: string | null;
  name: string;
  password: string;
}

export default class Content extends React.Component<Props, State> {
  public state: State = {
    email: "",
    errorMessage: null,
    name: "",
    password: "",
  };

  public render () {
    return (
      <Section className="SignupForm">
        <form onSubmit={this.handleFormSubmit}>
          <h2>Signup</h2>
          Already have an account? <Link to="/login">Click here to log in.</Link>
          <hr />
          {this.state.errorMessage && <Alert>{this.state.errorMessage}</Alert>}
          {this.renderNameInput()}
          {this.renderEmailInput()}
          {this.renderPasswordInput()}
          <Button type="submit">Submit</Button>
        </form>
      </Section>
    );
  }

  private handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      email: event.target.value,
    });
  }

  private handleFormSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    try {
      await this.props.signup({
        variables: {
          email: this.state.email,
          name: this.state.name,
          password: this.state.password,
        },
      });
    } catch (error) {
      this.setState({
        email: "",
        errorMessage: error.message,
      });
    }
  }

  private handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      name: event.target.value,
    });
  }

  private handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      password: event.target.value,
    });
  }

  private renderEmailInput () {
    return (
      <Input
        id="signup-email"
        label="Email Address"
        name="email"
        onChange={this.handleEmailChange}
        required={true}
        type="email"
        value={this.state.email}
      />
    );
  }

  private renderNameInput () {
    return (
      <Input
        autoFocus={true}
        id="signup-name"
        label="Name"
        name="name"
        onChange={this.handleNameChange}
        required={true}
        value={this.state.name}
      />
    );
  }

  private renderPasswordInput () {
    return (
      <Input
        id="signup-password"
        label="Password"
        name="password"
        onChange={this.handlePasswordChange}
        required={true}
        type="password"
        value={this.state.password}
      />
    );
  }
}
