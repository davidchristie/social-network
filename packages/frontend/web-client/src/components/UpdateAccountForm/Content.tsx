import { Button, Input, Section } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import {
  UpdateAccountData,
  UpdateAccountVariables,
} from "../../mutations/UpdateAccount";

interface Props {
  account: {
    email: string;
  };
  updateAccount: MutationFn<UpdateAccountData, UpdateAccountVariables>;
}

interface State {
  email: string;
}

export default class UpdateAccountForm extends React.Component<Props, State> {
  public state: State = {
    email: this.props.account.email,
  };

  public render () {
    const { updateAccount } = this.props;
    const { email } = this.state;
    return (
      <Section className="UpdateAccountForm">
        <h2>Account</h2>
        <form
          onSubmit={(event: React.FormEvent) => {
            event.preventDefault();
            updateAccount({
              variables: {
                data: {
                  email,
                },
              },
            });
          }}
        >
          <Input
            id="update-account-email"
            label="Email Address"
            name="email"
            onChange={this.emailChanged}
            required={true}
            type="email"
            value={this.state.email}
          />
          <Button type="submit">Save</Button>
        </form>
      </Section>
    );
  }

  private emailChanged = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      email: event.target.value,
    });
  }
}
