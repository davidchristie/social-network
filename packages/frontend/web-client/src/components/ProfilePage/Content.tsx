import { Avatar, Container, Section } from "design-system";
import React from "react";
import CreatePostForm from "../CreatePostForm";
import FollowProfileButton from "../FollowProfileButton";
import ProfilePosts from "../ProfilePosts";

export interface Props {
  account: {
    profile: {
      id: string;
    };
  } | null;
  profile: {
    avatar: {
      url: string;
    } | null;
    id: string;
    name: string;
  };
}

const Content: React.SFC<Props> = ({ account, profile }) => {
  const isOwnProfile = account && account.profile.id === profile.id;
  return (
    <Container>
      <Section>
        <Avatar
          image={profile.avatar ? profile.avatar.url : undefined}
          size="large"
        />
        <h1>{profile.name}</h1>
        {!isOwnProfile && (
          <FollowProfileButton profileId={profile.id} />
        )}
      </Section>
      <Section>
        {isOwnProfile && (
          <CreatePostForm />
        )}
        <ProfilePosts profileId={profile.id} />
      </Section>
    </Container>
  );
};

export default Content;
