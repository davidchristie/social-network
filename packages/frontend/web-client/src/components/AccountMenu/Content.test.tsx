import { Avatar, Menu } from "design-system";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { describeWithProps } from "test-utilities/react";
import Content, { Props, State } from "./Content";

const propsWithoutAvatar: Props = {
  account: {
    profile: {
      avatar: null,
      id: "profile_id",
    },
  },
};
const propsWithAvatar: Props = {
  account: {
    profile: {
      avatar: {
        url: "https://avatar_url.com",
      },
      id: "profile_id",
    },
  },
};

describe("Content component", () => {
  describeWithProps(
    "renders profile with avatar",
    Content,
    propsWithAvatar
  );

  describeWithProps(
    "renders profile without avatar",
    Content,
    propsWithoutAvatar
  );

  describe("when avatar is clicked", () => {
    let wrapper: ShallowWrapper<{}, State>;

    beforeEach(() => {
      const props = {
        account: {
          __typename: "Account",
          email: "account@email.com",
          id: "account_id",
          name: "Account",
          profile: {
            __typename: "Profile",
            avatar: null,
            id: "profile_id",
            name: "Profile",
          },
        },
      };
      wrapper = shallow(<Content {...props} />);
      wrapper.find(Avatar).simulate("click", {
        currentTarget: wrapper.find(Avatar).first().getElement(),
      });
    });

    it("opens menu", () => {
      const menu = wrapper.find(Menu);
      expect(menu.props().anchorElement).not.toBeNull();
    });

    describe("when menu is closed", () => {
      beforeEach(() => {
        const menu = wrapper.find(Menu);
        menu.props().onClose!();
      });

      it("updates state", () => {
        expect(wrapper.state().anchorElement).toBe(null);
      });
    });
  });
});
