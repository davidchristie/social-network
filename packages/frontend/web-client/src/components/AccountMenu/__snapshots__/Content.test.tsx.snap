// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component renders profile with avatar matches snapshot 1`] = `
<div>
  <Avatar$2
    image="https://avatar_url.com"
    onClick={[Function]}
    size="small"
  />
  <Menu$2
    anchorElement={null}
    onClose={[Function]}
    open={false}
  >
    <ButtonLink
      onClick={[Function]}
      to="profile/profile_id"
    >
      Profile
    </ButtonLink>
    <ButtonLink
      onClick={[Function]}
      to="/account"
    >
      Account
    </ButtonLink>
    <hr />
    <LogoutButton />
  </Menu$2>
</div>
`;

exports[`Content component renders profile without avatar matches snapshot 1`] = `
<div>
  <Avatar$2
    onClick={[Function]}
    size="small"
  />
  <Menu$2
    anchorElement={null}
    onClose={[Function]}
    open={false}
  >
    <ButtonLink
      onClick={[Function]}
      to="profile/profile_id"
    >
      Profile
    </ButtonLink>
    <ButtonLink
      onClick={[Function]}
      to="/account"
    >
      Account
    </ButtonLink>
    <hr />
    <LogoutButton />
  </Menu$2>
</div>
`;
